src/index.ts

```diff
@@ -1,5 +1,6 @@
 export { default as and } from './and';
 export { default as comparison } from './comparison';
+export { Comparison } from './comparison';
```

Comment:
Do we really need to expose it? As I see from examples, for custom operations it isn't necessary.

Reply:
In my opinion, yes.

The example from README:

```
import { and, comparison, eq, ge, inList, or } from "rsql-builder";

const query = and(
  comparison("genres", inList("sci-fi", "action", "non fiction")),
  or(
    comparison("director", eq("Christopher Nolan")),
    comparison("actor", eq("*Bale"))
  ),
  comparison("year", ge(2000))
);
```

could be written as:
```
import { and, comparison, eq, ge, inList, or } from "rsql-builder";

const comparison1: Comparison =   comparison("genres", inList("sci-fi", "action", "non fiction"));
const comparison2: Comparison = comparison("director", eq("Christopher Nolan"));
const comparison3: Comparison = comparison("actor", eq("*Bale"))
comparison4: Comparison = comparison("year", ge(2000));

const query = and(
  comparison1,
  or(comparison2, comparison3),
  comparsion4
);
```

which is much cleaner when you have complex queries and/or conditional comparisons.

Reply:
```ts
const comparison1: Comparison =   comparison("genres", inList("sci-fi", "action", "non fiction"));
```
But in this case, adding type to `comparison1` is redundant, due to the type will be inferred automatically.

Reply:
Correct, but in my opinion, if a user can create a `Comparison` instance, he should be able to have `Comparison` type as well.

example:
```
let conditionalQuery: Comparison;
const inListQuery: Comparison = comparison("genres", inList("sci-fi", "action", "non fiction"));
const notInListQuery: Comparison = comparison("genres", outList("sci-fi", "action", "non fiction"));
const defaultQuery: Comparison = comparison("actor", eq("*Bale"));

if (something) {
  conditionalQuery = inListQuery;
} else if (something2) {
  conditionalQuery = notInListQuery;
} else {
  conditionalQuery = defaultQuery;
}

const query = and(
  conditionalQuery,
  comparison("year", ge(2000))
);
```

Of course, this can be written in a cleaner way, but you get the point.

Anyway, the decision is up to you.

Reply:
For time being `Comparison` isn't just a type it's a class. If we'll expose, I think it could lead to the wrong usage of the library. And devs start using it instead of factory `comparison`, which is more simple and clear for my taste. So, I would prefer to keep `Comparison` class kind of internal.

Reply:
I get it. Everything clear, let's keep `Comparison` class internal then.
I've updated the pull request.